bip-musig2/reference.py

```diff
@@ -159,11 +159,13 @@ def cpoint_extended(x: bytes) -> Optional[Point]:
     else:
         return cpoint(x)
 
-KeyGenContext = namedtuple('KeyGenContext', ['Q', 'gacc', 'tacc'])
+KeyGenContext = namedtuple('KeyGenContext', ['Q', 'tacc'])
+XOnlyKeyGenContext = namedtuple('XOnlyKeyGenContext', ['Q', 'gacc', 'tacc'])
 
-def get_pk(keygen_ctx: KeyGenContext) -> bytes:
-    Q, _, _ = keygen_ctx
-    return bytes_from_point(Q)
+# HACK: return the public key for either context
+def get_pk(keygen_ctx):
+    Q, *_ = keygen_ctx
+    return Q
```

Comment:
I created two functions and stored the full point in both contexts. I prefered dropping the y-coordinate and serializing the x coordinate simply because it means it makes storing an invalid representation of an even y coordinate impossible.